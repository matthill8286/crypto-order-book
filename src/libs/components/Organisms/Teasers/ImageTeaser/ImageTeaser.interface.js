
//# sourceMappingURL=ImageTeaser.interface.js.map