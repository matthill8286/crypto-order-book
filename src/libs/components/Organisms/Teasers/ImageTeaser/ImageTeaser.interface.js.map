{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"ImageTeaser.interface.js"}