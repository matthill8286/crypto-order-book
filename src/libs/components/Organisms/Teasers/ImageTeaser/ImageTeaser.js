import React from 'react';
import { Card } from "../../../Atoms/Card";
import { Picture } from "../../../Atoms/Picture";
export var ImageTeaser = function ImageTeaser(_ref) {
  var badges = _ref.badges,
      badgeActionType = _ref.badgeActionType,
      _ref$elevation = _ref.elevation,
      elevation = _ref$elevation === void 0 ? 0 : _ref$elevation,
      _ref$elevationHover = _ref.elevationHover,
      elevationHover = _ref$elevationHover === void 0 ? 4 : _ref$elevationHover,
      finalImages = _ref.finalImages;
  return React.createElement(Card, {
    elevation: elevation,
    elevationHover: elevationHover,
    shape: "rounded-small",
    display: "flex",
    fullHeight: true,
    badges: badges,
    badgeActionType: badgeActionType
  }, React.createElement(Picture, {
    src: finalImages[0],
    srcLg: finalImages[1],
    srcMd: finalImages[2],
    srcSm: finalImages[3],
    objectFit: "cover",
    height: "100%",
    width: "100%"
  }));
};
//# sourceMappingURL=ImageTeaser.js.map