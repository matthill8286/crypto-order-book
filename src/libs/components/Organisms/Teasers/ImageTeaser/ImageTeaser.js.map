{"version":3,"sources":["../../../../../src/components/Organisms/Teasers/ImageTeaser/ImageTeaser.tsx"],"names":["React","Card","Picture","ImageTeaser","badges","badgeActionType","elevation","elevationHover","finalImages"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,IAAT;AACA,SAASC,OAAT;AAGA,OAAO,IAAMC,WAAyC,GAAG,SAA5CA,WAA4C,OAMnD;AAAA,MALJC,MAKI,QALJA,MAKI;AAAA,MAJJC,eAII,QAJJA,eAII;AAAA,4BAHJC,SAGI;AAAA,MAHJA,SAGI,+BAHQ,CAGR;AAAA,iCAFJC,cAEI;AAAA,MAFJA,cAEI,oCAFa,CAEb;AAAA,MADJC,WACI,QADJA,WACI;AACJ,SACE,oBAAC,IAAD;AACE,IAAA,SAAS,EAAEF,SADb;AAEE,IAAA,cAAc,EAAEC,cAFlB;AAGE,IAAA,KAAK,EAAC,eAHR;AAIE,IAAA,OAAO,EAAC,MAJV;AAKE,IAAA,UAAU,MALZ;AAME,IAAA,MAAM,EAAEH,MANV;AAOE,IAAA,eAAe,EAAEC;AAPnB,KAQE,oBAAC,OAAD;AACE,IAAA,GAAG,EAAEG,WAAW,CAAC,CAAD,CADlB;AAEE,IAAA,KAAK,EAAEA,WAAW,CAAC,CAAD,CAFpB;AAGE,IAAA,KAAK,EAAEA,WAAW,CAAC,CAAD,CAHpB;AAIE,IAAA,KAAK,EAAEA,WAAW,CAAC,CAAD,CAJpB;AAKE,IAAA,SAAS,EAAC,OALZ;AAME,IAAA,MAAM,EAAC,MANT;AAOE,IAAA,KAAK,EAAC;AAPR,IARF,CADF;AAoBD,CA3BM","sourcesContent":["import React from 'react'\nimport { Card } from '@/components/Atoms/Card'\nimport { Picture } from '@/components/Atoms/Picture'\nimport { ImageTeaserFCProps } from './ImageTeaser.interface'\n\nexport const ImageTeaser: React.FC<ImageTeaserFCProps> = ({\n  badges,\n  badgeActionType,\n  elevation = 0,\n  elevationHover = 4,\n  finalImages,\n}) => {\n  return (\n    <Card\n      elevation={elevation}\n      elevationHover={elevationHover}\n      shape=\"rounded-small\"\n      display=\"flex\"\n      fullHeight\n      badges={badges}\n      badgeActionType={badgeActionType}>\n      <Picture\n        src={finalImages[0]}\n        srcLg={finalImages[1]}\n        srcMd={finalImages[2]}\n        srcSm={finalImages[3]}\n        objectFit=\"cover\"\n        height=\"100%\"\n        width=\"100%\"\n      />\n    </Card>\n  )\n}\n"],"file":"ImageTeaser.js"}