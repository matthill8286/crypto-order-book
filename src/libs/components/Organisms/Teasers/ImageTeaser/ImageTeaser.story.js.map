{"version":3,"sources":["../../../../../src/components/Organisms/Teasers/ImageTeaser/ImageTeaser.story.tsx"],"names":["number","radios","storiesOf","React","StorybookRouterWrapper","ImageTeaser","getTransformedImageVersion","actionTypes","prominent","secondary","badges","id","name","actionType","module","add","knobs","badgeActionType","finalImages","height","range","min","max","step","width","display"],"mappings":";;AAAA,SAASA,MAAT,EAAiBC,MAAjB,QAA+B,wBAA/B;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AAEA,SAASC,sBAAT;AACA,SAASC,WAAT;AAEA,SAASC,0BAAT;AAEA,IAAMC,WAA+C,GAAG;AACtDC,EAAAA,SAAS,EAAE,WAD2C;AAEtDC,EAAAA,SAAS,EAAE;AAF2C,CAAxD;AAKA,IAAMC,MAAmB,GAAG,CAC1B;AAAEC,EAAAA,EAAE,EAAE,CAAN;AAASC,EAAAA,IAAI,EAAE;AAAf,CAD0B,EAE1B;AAAED,EAAAA,EAAE,EAAE,CAAN;AAASC,EAAAA,IAAI,EAAE,WAAf;AAA4BC,EAAAA,UAAU,EAAE;AAAxC,CAF0B,CAA5B;AAKAX,SAAS,CAAC,6CAAD,EAAgDY,MAAhD,CAAT,CAAiEC,GAAjE,CAAqE,SAArE,EAAgF,YAAM;AACpF,MAAMC,KAAK,GAAG,SAARA,KAAQ,GAA0B;AACtC,WAAO;AACLC,MAAAA,eAAe,EAAEhB,MAAM,CAAC,MAAD,EAASM,WAAT,EAAsBA,WAAW,CAAC,WAAD,CAAjC,CADlB;AAELW,MAAAA,WAAW,EAAE,CACXZ,0BAA0B,CAAC,sBAAD,CADf,EAEXA,0BAA0B,CAAC,sBAAD,CAFf,EAGXA,0BAA0B,CAAC,sBAAD,CAHf,EAIXA,0BAA0B,CAAC,sBAAD,CAJf;AAFR,KAAP;AASD,GAVD;;AAYA,MAAMa,MAAM,GAAGnB,MAAM,CAAC,QAAD,EAAW,GAAX,EAAgB;AACnCoB,IAAAA,KAAK,EAAE,IAD4B;AAEnCC,IAAAA,GAAG,EAAE,GAF8B;AAGnCC,IAAAA,GAAG,EAAE,GAH8B;AAInCC,IAAAA,IAAI,EAAE;AAJ6B,GAAhB,CAArB;AAOA,SACE,oBAAC,sBAAD,QACE;AACE,IAAA,KAAK,EAAE;AACLJ,MAAAA,MAAM,EAANA,MADK;AAELK,MAAAA,KAAK,EAAE,MAFF;AAGLC,MAAAA,OAAO,EAAE;AAHJ;AADT,KAME,oBAAC,WAAD;AAAa,IAAA,GAAG,EAAE,iBAAiBN;AAAnC,KAA+CH,KAAK,EAApD;AAAwD,IAAA,MAAM,EAAEN;AAAhE,KANF,CADF,CADF;AAYD,CAhCD","sourcesContent":["import { number, radios } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { BadgeActionType, BadgeType } from '@/components/Atoms/Badge/Badge.interface'\nimport { StorybookRouterWrapper } from '@/utils/StorybookWrapper'\nimport { ImageTeaser } from './ImageTeaser'\nimport { ImageTeaserFCProps } from './ImageTeaser.interface'\nimport { getTransformedImageVersion } from '@/utils/useGraphCmsImages'\n\nconst actionTypes: { [key: string]: BadgeActionType } = {\n  prominent: 'prominent',\n  secondary: 'secondary',\n}\n\nconst badges: BadgeType[] = [\n  { id: 0, name: 'Featured' },\n  { id: 1, name: 'Sponsored', actionType: 'prominent' },\n]\n\nstoriesOf('Design System/Organisms/Teasers/ImageTeaser', module).add('Default', () => {\n  const knobs = (): ImageTeaserFCProps => {\n    return {\n      badgeActionType: radios('Type', actionTypes, actionTypes['prominent']),\n      finalImages: [\n        getTransformedImageVersion('c8GLaIBVRa6EelN5baVr'),\n        getTransformedImageVersion('OWr5QtrXQ1eHiwMdrDM7'),\n        getTransformedImageVersion('fj7dbi1rSa2cGlaQqwDG'),\n        getTransformedImageVersion('NPgqQFqR0uJ1XhB00sGg'),\n      ],\n    }\n  }\n\n  const height = number('height', 300, {\n    range: true,\n    min: 100,\n    max: 600,\n    step: 1,\n  })\n\n  return (\n    <StorybookRouterWrapper>\n      <div\n        style={{\n          height,\n          width: '100%',\n          display: 'flex',\n        }}>\n        <ImageTeaser key={'ImageTeaser_' + height} {...knobs()} badges={badges} />\n      </div>\n    </StorybookRouterWrapper>\n  )\n})\n"],"file":"ImageTeaser.story.js"}