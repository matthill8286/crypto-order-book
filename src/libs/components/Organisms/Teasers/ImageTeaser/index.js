export * from "./ImageTeaser";
//# sourceMappingURL=index.js.map