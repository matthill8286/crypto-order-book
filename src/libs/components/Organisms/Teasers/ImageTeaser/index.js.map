{"version":3,"sources":["../../../../../src/components/Organisms/Teasers/ImageTeaser/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './ImageTeaser'\n"],"file":"index.js"}