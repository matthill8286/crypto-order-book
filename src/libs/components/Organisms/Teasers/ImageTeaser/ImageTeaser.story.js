function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { number, radios } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { StorybookRouterWrapper } from "../../../../utils/StorybookWrapper";
import { ImageTeaser } from "./ImageTeaser";
import { getTransformedImageVersion } from "../../../../utils/useGraphCmsImages";
var actionTypes = {
  prominent: 'prominent',
  secondary: 'secondary'
};
var badges = [{
  id: 0,
  name: 'Featured'
}, {
  id: 1,
  name: 'Sponsored',
  actionType: 'prominent'
}];
storiesOf('Design System/Organisms/Teasers/ImageTeaser', module).add('Default', function () {
  var knobs = function knobs() {
    return {
      badgeActionType: radios('Type', actionTypes, actionTypes['prominent']),
      finalImages: [getTransformedImageVersion('c8GLaIBVRa6EelN5baVr'), getTransformedImageVersion('OWr5QtrXQ1eHiwMdrDM7'), getTransformedImageVersion('fj7dbi1rSa2cGlaQqwDG'), getTransformedImageVersion('NPgqQFqR0uJ1XhB00sGg')]
    };
  };

  var height = number('height', 300, {
    range: true,
    min: 100,
    max: 600,
    step: 1
  });
  return React.createElement(StorybookRouterWrapper, null, React.createElement("div", {
    style: {
      height: height,
      width: '100%',
      display: 'flex'
    }
  }, React.createElement(ImageTeaser, _extends({
    key: 'ImageTeaser_' + height
  }, knobs(), {
    badges: badges
  }))));
});
//# sourceMappingURL=ImageTeaser.story.js.map