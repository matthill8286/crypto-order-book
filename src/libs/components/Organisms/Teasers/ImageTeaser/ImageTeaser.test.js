import React from 'react';
import { renderWithThemeAndRouter } from "../../../../testRenderer";
import { ImageTeaser } from "./ImageTeaser";
var images = ['public/images/teasers/500x560.jpg', 'public/images/teasers/700x560.jpg', 'public/images/teasers/1000x360.jpg', 'public/images/teasers/1200x360.jpg'];
describe('ImageTeaser', function () {
  it('renders correctly: default teaser with badges', function () {
    var tree = renderWithThemeAndRouter(React.createElement(ImageTeaser, {
      badges: [{
        id: 0,
        name: 'Featured'
      }, {
        id: 1,
        name: 'Live@@@'
      }],
      badgeActionType: "prominent",
      finalImages: images
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=ImageTeaser.test.js.map