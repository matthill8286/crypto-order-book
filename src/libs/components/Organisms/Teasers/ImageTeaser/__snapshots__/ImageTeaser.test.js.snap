// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageTeaser renders correctly: default teaser with badges 1`] = `
<div
  className="Cardstyled__StyledCardWrapper-sc-1gxmhl7-6 gyZXAf"
>
  <div
    className="Cardstyled__StyledBadgeLineWrapper-sc-1gxmhl7-0 cFLGQx"
  >
    <div
      className="Badgestyled__StyledBadgeLineWrapper-sc-1xo0r7x-1 ejODHn"
    >
      <div
        className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
      >
        <span
          className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
          fontSize="xxs"
        >
          <div
            className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
          >
            Featured
          </div>
        </span>
      </div>
      <div
        className="Badgestyled__StyledBadgeWrapper-sc-1xo0r7x-2 TMGmv"
      >
        <span
          className="Typostyled__StyledInfoTypo-sc-1o9a6sy-0 hycfbo Badgestyled__StyledTypo-sc-1xo0r7x-0 ePrcMg"
          fontSize="xxs"
        >
          <div
            className="Badgestyled__StyledOverflow-sc-1xo0r7x-3 iyMzXH"
          >
            Live@@@
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="Cardstyled__StyledCard-sc-1gxmhl7-1 sFHjI"
    display="flex"
    elevation={0}
    shape="rounded-small"
  >
    <div
      className="Picturestyled__StyledPicture-sc-18jsup7-0 jIsvqb"
      height="100%"
      width="100%"
    >
      <picture
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <source
          media="(max-width: 511px)"
          srcSet="public/images/teasers/1200x360.jpg"
        />
        <source
          media="(max-width: 751px)"
          srcSet="public/images/teasers/1000x360.jpg"
        />
        <source
          media="(max-width: 1007px)"
          srcSet="public/images/teasers/700x560.jpg"
        />
        <img
          alt="image"
          height="100%"
          loading="eager"
          src="public/images/teasers/500x560.jpg"
          width="100%"
        />
      </picture>
    </div>
  </div>
</div>
`;
